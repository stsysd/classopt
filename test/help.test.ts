import {
  Arg,
  Cmd,
  Command,
  Help,
  help,
  Name,
  Opt,
  Rest,
  Version,
} from "../mod.ts";
import { assertSnapshot } from "https://deno.land/std@0.214.0/testing/snapshot.ts";

Deno.test("snapshot help", async (t) => {
  class Program {
    @Opt({ type: "string", short: "s", long: "str" })
    foo = "foo";

    @Arg({ about: "about bar arg" })
    bar!: string;

    @Arg({ optional: true })
    baz?: string;

    @Rest({ about: "about rest args" })
    rest!: string[];
  }

  await assertSnapshot(t, help(new Program()));
});

Deno.test("snapshot help (Command)", async (t) => {
  @Name("hoge")
  @Version("1.0.0")
  @Help("this is a hoge program")
  class Program extends Command {
    @Opt({ type: "string", short: "s", long: "str" })
    foo = "foo";

    @Arg({ about: "about bar arg" })
    bar!: string;

    @Arg({ optional: true })
    baz?: string;

    @Rest({ about: "about rest args" })
    rest!: string[];

    execute() {
      // do nothing
    }
  }

  await assertSnapshot(t, help(new Program()));
});

Deno.test("snapshot help (sub)", async (t) => {
  @Name("hoge")
  class Sub {
  }

  class Program {
    @Opt({ type: "string", short: "s", long: "str" })
    foo = "foo";

    @Cmd(Sub)
    sub?: Sub;
  }

  await assertSnapshot(t, help(new Program()));
});
