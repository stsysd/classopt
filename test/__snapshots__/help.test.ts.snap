export const snapshot = {};

snapshot[`snapshot help 1`] = `
"program
USAGE
    program [OPTIONS] <bar> [baz] [rest]...

OPTIONS
    -s, --str <string>    

ARGS
    <bar>        about bar arg
    [baz]        
    [rest]...    about rest args"
`;

snapshot[`snapshot help (Command) 1`] = `
"hoge - this is a hoge program
USAGE
    hoge [OPTIONS] <bar> [baz] [rest]...

OPTIONS
    -s, --str <string>    
    -h, --help            Prints help information
    -V, --version         Prints version information

ARGS
    <bar>        about bar arg
    [baz]        
    [rest]...    about rest args"
`;

snapshot[`snapshot help (sub) 1`] = `
"program
USAGE
    program [OPTIONS] <SUBCOMMAND>

OPTIONS
    -s, --str <string>    

COMMANDS
    hoge    "
`;
